core/context.go

```diff
@@ -26,17 +26,30 @@ import (
 	"go.uber.org/fx/service"
 )
 
-// Context embedds Host and go context for use
+// Context embeds Host and go context for use
 type Context interface {
 	gcontext.Context
 	service.Host
+	WithContext(ctx gcontext.Context) *context
```

Comment:
I was following the existing file's convention but tried this out just now. It causes an interface type loop since we have a struct named context in the same file.